Prepare for rust 1.62.1

skunert opened the pull request:
- update UI tests for rust 1.62.1

alvicsam commented:
clippy is not happy with the latest nightly https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/1695838

skunert commented:
I had to replace some automatic derives of `scale_info::TypeInfo`. I introduced manual implementations for the two warnings we saw (implementation is `cargo-expand` output mostly) and fixed the warnings.

Also clippy should be fixed.

bkchr commented on primitives/runtime/src/bounded/bounded_vec.rs:
```diff
@@ -104,10 +104,45 @@ where
+// This can be replaced with
+// #[derive(scale_info::TypeInfo)]
+// #[scale_info(skip_type_params(S))]
+// again once this issue is fixed in the rust compiler: https://github.com/rust-lang/rust/issues/96956
```
Please create an issue for this.

bkchr replied:
So, that we don't forget it.

skunert replied:
https://github.com/paritytech/substrate/issues/11915, I will also put it in the comment here
